java/src/main/java/ai/rapids/cudf/ColumnViewPointerAccess.java

```diff
@@ -0,0 +1,47 @@
+/*
+ *
+ *  Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package ai.rapids.cudf;
+
+//Evolving and Unstable
+
+/**
+ * Interface that provides access only methods to cudf column_views
+ */
+public interface ColumnViewPointerAccess<T> {
+
+  long getColumnView();
+
+  ColumnViewPointerAccess<T> getChildColumnView(int childIndex);
+
+  T getDataBuffer();
+
+  T getOffsetBuffer();
+
+  T getValidityBuffer();
+
+  long getNullCount();
+
+  DType getDataType();
+
+  long getNumRows();
+
+  int getNumChildren();
+
+  void close();
```

Comment:
Can we make this `AutoCloseable` instead?

Reply:
Sure. I had it that way but I am curiosu what to do in copyToHost if we do have an exception? deleteCudfColumnView is throwing a runtime error so should be throw a cudf exception or have copyToHost throw it.

Reply:
Having it be part of copyToHost signature makes for a lot of test code change.

Reply:
This is addressed.